app/controllers/promoting.controller.js

```diff
@@ -36,15 +36,44 @@ exports.userIsPromoted = async (req, res, next) => {
     .catch((err) => next(err));
 };
 
+// Get users that LOGGED IN user promotes
+exports.sessionPromotes = (req, res, next) => {
+  const { user } = req;
+
+  user
+    .promotes()
+    .then((promotions) => {
+      return res.status(200).send({
+        message: 'Promotions sent successfully',
+        data: promotions,
+      });
+    })
+    .catch((err) => next(err));
+};
+
+// Get users that LOGGED IN user is promoted by
+exports.sessionPromoted = (req, res, next) => {
+  const { user } = req;
+
+  user
+    .promotedBy()
+    .then((promoters) => {
+      return res.status(200).send({
+        message: 'Promoters sent successfully',
+        data: promoters,
+      });
+    })
+    .catch((err) => next(err));
+};
+
 // Create entry in Promoting
-exports.promotingOneUser = async (req, res, next) => {
+exports.promotingOneUser = (req, res, next) => {
   const {
-    body: { userId },
     params: { promotedId },
   } = req;
 
   User.findById(promotedId)
-    .then(() => Promoting.createPromotion(userId, promotedId))
+    .then(() => Promoting.createPromotion(req.user.id, promotedId))
```

Comment:
The idea behind passport is when a user is logged in makes a request, they also send a cookie. The cookie comes in and passport compares it to the session id associated with the logged in user. If everything matches then passport loads the `user` object in the `req` object automatically - you can see the function that makes it happen in the `app/passport/setup.js` file, `deserializeUser` (that is why you sometimes see me using `req.user`, that's literally the user object tied to the logged in user, already fetched from the db automatically and put in the request by passport.

That said, since only the logged in user should be able to create or delete a promotion using their own id, using the user object loaded from the req object makes sense and makes it so we don't need to pass the userId in every request anymore. It also adds an additional layer of security and it's less work for the frontend because they now don't need to send the `userId` anymore, since we know which user the request is coming from. This is valid for your two other comments too. 

But you made me notice that I should have destructured the id from the user object, which I'm going to do now :)

Reply:
Ok, thanks for the explanation! :)